renku/models/_jsonld.py

```diff
@@ -375,8 +381,10 @@ def from_jsonld(
 
         data_ = {}
         if client:
-            data_['client'] = client
-            data_['commit'] = commit
+            if client:
+                data_['client'] = client
+            if commit:
+                data_['commit'] = commit
```

Comment:
Maybe we should add a comment in which cases commit could be None? Not sure if we have that comment anywhere, could be useful for a future reference.